manifests/indexer.pp

```diff
@@ -134,6 +135,14 @@
     }
   }
 
+  if $full_indexer_reinstall {
+    exec { "Clean security init lockfile":
+      path    => '/usr/bin:/bin',
+      command => "rm -f ${indexer_security_init_lockfile}",
+      require => Package['wazuh-indexer'],
+    }
```

Comment:
Use an exec resource only when you really need to and ensure that the cleanup is happend before the indexer security init run.
```suggestion
    file { $indexer_security_init_lockfile:
      ensure => absent,
      require => Package['wazuh-indexer'],
      before  => Exec['Initialize the Opensearch security index in Wazuh indexer'],
    }
```